### ISSUE TITLE
Converting from QtSingleApplication causing crash

### ISSUE THREAD

**Issue by jonaski** [itay-grudev/SingleApplication#61]
I'm trying to replace QtSingleApplication with this, however it crashes, no idea why.
It crashes here:
https://github.com/jonaski/strawberry/blob/singleapplication/src/core/iconloader.cpp#L51
within Qt at: QPixmap::fromImage()
I'm using Qt 5.12
See https://github.com/jonaski/strawberry/pull/40
Replacing SingleApplication with QApplication or QtSingleApplication the crash is gone.
Any suggestions?

**Comment by itay-grudev** [comment]
I am not sure what your logic behind this is:

https://github.com/jonaski/strawberry/blob/1e0a450169d17b4db4451528a92fd48d6bd1fe89/src/main.cpp#L126

But could it end up initialising QApplication twice? The second call is here:

https://github.com/jonaski/strawberry/blob/1e0a450169d17b4db4451528a92fd48d6bd1fe89/src/main.cpp#L165

Additionally where is the `IconLoader::Load` called from?

**Comment by jonaski** [comment]
The first initialization of SingleApplication is not in use, since its within the #if 0, it was used with QtSingleApplication like that as a CoreApplication class, and I just renamed the class and functions before I skipped it with the #if 0.

The call to IconLoader::Load is here:
https://github.com/jonaski/strawberry/blob/singleapplication/src/collection/collectiondirectorymodel.cpp#L43
But IconLoader is used everywhere and is crashing all over the place.
I can skip it, but then it crashes when creating the system tray icon.
It is again initialized through SCollection through the Application class which is done before the mainwindow, maybe that somehow causes problems.
Here: https://github.com/jonaski/strawberry/blob/singleapplication/src/main.cpp#L212

**Comment by jonaski** [comment]
Yes

**Comment by jonaski** [comment]
I reverted back until I found out that it crashed after this commit:
https://github.com/itay-grudev/SingleApplication/commit/6b78aefd28dcbf83a492e35e19b7d99b526e5082
So I found out that just replacing the macro QAPPLICATION_CLASS with QApplication it does not crash anymore.
I don't understand why because i'm defining QAPPLICATION_CLASS before I include singleapplication.h
Maybe it has to do with some of the headers I include before that. I will investigate a bit more.
